modules/tinymce/src/core/main/ts/SelectionOverrides.ts

```diff
@@ -160,6 +160,15 @@ const SelectionOverrides = (editor: Editor): SelectionOverrides => {
       }
     });
 
+    editor.on('NodeChange', (e) => {
```

Comment:
Aren't there other events already used in this file that would be more appropriate? This will run a lot more often than when tab is pressed.

Or can we add a simple "is fake caret set" check to reduce how often this code does anything significant? It could probably be a wrapper around `lastVisualCaret.isSet()` 🤔

Reply:
Yeah it would need to be keypress or similar. This will do fine, thank you

Reply:
I tried to implement it in the [keypress](https://github.com/tinymce/tinymce/blob/main/modules/tinymce/src/core/main/ts/SelectionOverrides.ts#L138) of this file, but it seems prevented in this case and I don't know why :sweat_smile:

Reply:
I think this should run on `focusin` that is fired when the focus is moved from outside the element to inside the element. 

This seems to fix it. But my guess is that parts of the SelectionOverride maybe `AfterSetSelectionRange` is the code that needs to be executed on `focusin` so that it re-freshes the fake caret state. Needs some digging.
```
editor.on('focusin',  () => {
    editor.selection.setRng(editor.selection.getRng());
});
```

Reply:
I changed the event from `NodeChange` to `focusin` but using `editor.selection.setRng(editor.selection.getRng())` instead of the current code, solve this bug, but it breaks many tests, so I would stay with the current code but on `focusin` instead of `NodeChange`